core/services/deploykey/deploykey.go

```diff
@@ -0,0 +1,58 @@
+package deploykey
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/weaveworks/weave-gitops/core/adapters/fluxbin"
+	"github.com/weaveworks/weave-gitops/pkg/gitproviders"
+	"github.com/weaveworks/weave-gitops/pkg/models"
+	"github.com/weaveworks/weave-gitops/pkg/services/auth"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type Manager interface {
+	Create(ctx context.Context, name types.NamespacedName, repoURL gitproviders.RepoURL, token string) (models.GeneratedSecretName, error)
+}
+
+func NewManager(k8s client.Client, rt http.RoundTripper) Manager {
+	return manager{k8s: k8s, http: rt}
+}
+
+type manager struct {
+	k8s  client.Client
+	http http.RoundTripper
```

Comment:
It looks ok to me. I think it could be useful for integration tests. Or even cache mechanisms to speed them up. This is what I used a while ago when I cached the Github api interactions to speed unit tests/integration tests up.